### ISSUE TITLE
how to stacking with  Keras model

### ISSUE THREAD

**Issue by oras903** [rasbt/mlxtend#755]
#### Describe the workflow you want to enable

I want to use  StackingCVClassifier  to  stack   sklearn model  and  keras model ,  and  I have done some work ,    but when  I  try  sklearn's  cross_val_score    method  ,  I  get " can't pickle _thread.RLock objects"  error message.

#### Describe your proposed solution
 I  found  the  sklearn's   cross_val_score / gridsearchcv  method  use  clone()  or  someother  method  that  not  support   Keras model

 Is there a way  that  adapt Keras model to  sklearn  structure  and  benefit from StackingCVClassifier ?   especialy  GridSearchCV    

#### Additional context

**Comment by qiagu** [comment]
@oras903 You may want to have a look at https://github.com/goeckslab/Galaxy-ML.
Keras wrapper was reimplemented for better compatibility with sklearn APIs. No guarantee it works. Please let me know what you think.

**Comment by rasbt** [comment]
Haven't used it myself, but year, this looks like it could get the job done:

```python
from galaxy_ml.keras_galaxy_models import KerasGClassifier


# build a DNN classifier
model = Sequential()
model.add(Dense(64))
model.add(Activation(‘relu'))
model.add((Dense(1, activation=‘sigmoid’)))
config = model.get_config()

classifier = KerasGClassifier(config, random_state=42)
...
```

from their Readme, it also looks like they support the stacking classes in MLxtend explicitly, too. Maybe adding this to the stacking models' documentations would be nice.

**Comment by qiagu** [comment]
Yes. Galaxy-ML supports stacking classes in MLxtend well. A demo tool can be accessible at https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_stacking_ensemble_models/sklearn_stacking_ensemble_models/1.0.8.2 ,
where MLxtend classes are grouped under the `Choose the stacking ensemble type`. Together with other tools, like [Pipeline Builder](https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_build_pipeline/sklearn_build_pipeline/1.0.8.2)  and [Hyperparameter Search](https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Fsklearn_searchcv%2Fsklearn_searchcv%2F1.0.8.2), decent machine learning work can be done on the web.

Tell a story here. We invested a lot time in making complex modeling, like stacking, easier in Galaxy-ML tools. I believe we did good job so far. However, when practicing in the biology field, complex modeling doesn't always justify its benefit. Biology people often prefer a simple model with better interpretability.
